src/inference.jl

```diff
@@ -96,6 +96,10 @@ RunState(c::ModelConfig) = RunState(;
     logits         = zeros(Float32, c.vocab_size),
     kvcache_layers = [KVCache(c.dim ÷ c.n_heads, c.n_kv_heads, c.seq_len) for _ in 1:c.n_layers],
 )
+get_run_state(model::LanguageModel) = RunState(model.config)
+
+# A nicer silu could be? silu(x) = x*σ(x)
+silu(x) = x * (1f0 / (1f0 + exp(-x)))
```

Comment:
This should probably be 

```suggestion
silu(x::T) where {T<:Real} = x * (one(T) / (one(T) + exp(-x)))
```

because this will promote `Float16` to `Float32`:

```julia-repl
julia> @code_warntype silu(Float16(1))
MethodInstance for silu(::Float16)
  from silu(x) @ Main REPL[1]:2
Arguments
  #self#::Core.Const(silu)
  x::Float16
Body::Float32
1 ─ %1 = -x::Float16
│   %2 = Main.exp(%1)::Float16
│   %3 = (1.0f0 + %2)::Float32
│   %4 = (1.0f0 / %3)::Float32
│   %5 = (x * %4)::Float32
└──      return %5
```

Reply:
It is now going to be this: `silu(x) = x*σ(x)` 
I am not sure if it is the same. :thinking: